### ISSUE TITLE
[MNG-336] support "+" in artifact, group id's

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#2633]
**[John Dennis Casey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jdcasey)** opened **[MNG-336](https://issues.apache.org/jira/browse/MNG-336?redirect=false)** and commented

See MNG-323.

This is not a repoclean issue, but since the issue filed originally is a subtask of repoclean-related things, I'm creating a new instance of the issue here.

The reason stax-1.1.1-dev wasn't being propagated to the m2 repository was unrelated to this, and has been resolved (it's related to the problems we were having with commons-logging-1.0.4)

Basically, Dan's contention is that we need to support the '+' character in POM groupId's and artifactId's.

I'm not taking a position on this (not sure I agree that this is a good idea), only relaying the info for more general consumption. I don't want this issue to lose visibility as a subtask of a repoclean cleanup issue...

---

**Affects:** 2.0-alpha-1

**Comment by jira-importer** [comment]
**[Daniel Diephouse](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dandiep)** commented

Just to be clear, my contention is that when parsing M1 POMs and converting them to M2 POMs you should convert from the + syntax to the new & improved artifactId/groupId format.  I'm not suggesting that you support the '+' syntax in M2 poms.

**Comment by jira-importer** [comment]
**[John Dennis Casey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jdcasey)** commented

my misunderstanding. I'm copying your latest comment back to the original issue, and we'll pursue from there.

**Comment by jira-importer** [comment]
**[Brett Porter](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=brettporter)** commented

reopening to adjust fix for
